### PR TITLE
fix(aws): Changes refresh methodology

### PR DESCRIPTION
**Pull request by thomastaylor312**
Due to some issues with AWS auth, this changes how `Refresh` works.

### CODE REVIEW

File: auth/aws.go [hunk 1]
```diff
@@ -143,51 +148,53 @@ func (a *AWSAuth) GetToken(f *os.File) (string, error) {
...
-	return nil
+	// A note on why we are just reauthenticating: You can refresh an AWS token,
+	// but there is a limit (24) to the number of refreshes and the API requests
+	// that you refresh your token on every SDB creation. When doing this in an
+	// automation context, you could surpass this limit. You could not refresh
+	// the token, but it can get you in to a state where you can't perform some
+	// operations. This is less than ideal but better than having an arbitary
+	// bound on the number of refreshes and having to track how many have been
+	// done.
+	return a.authenticate()
```
Comment (rji) [inline auth/aws.go]: 👍

Comment (fieldju) [inline auth/aws.go]: So you may wait till this code is actually in CMS.
The limit is going to be configurable and not guaranteed to be 24.

The limit and the current count will be returned as part of the metadata in a token.

Comment (fieldju) [inline auth/aws.go]: But re-authenticating is a better option for non user driven flows anyways, which is what almost all the other clients do.

Comment (thomastaylor312) [inline auth/aws.go]: Yeah, that was why I switched this over @fieldju. I didn't want to run into any weird issues with the number of refreshes

**Pull request merged**